flocker/node/agents/ebs.py

```diff
@@ -128,15 +106,18 @@ def _run_with_logging(self, *args, **kwargs):
         """
         original = getattr(self, original_name)
         method = getattr(original, method_name)
-        try:
-            return method(*args, **kwargs)
-        except EC2ResponseError as e:
-            BOTO_EC2RESPONSE_ERROR(
-                code=e.code,
-                message=e.message,
-                request_id=e.request_id,
-            ).write()
-            raise
+
+        # Trace IBlockDeviceAPI ``method`` as Eliot Action.
+        with AWS_ACTION(operation=[method_name, args, kwargs]):
+            try:
+                return method(*args, **kwargs)
+            except EC2ResponseError as e:
+                BOTO_EC2RESPONSE_ERROR(
+                    aws_code=e.code,
+                    aws_message=e.message,
+                    aws_request_id=e.request_id,
+                ).write()
```

Comment:
It'd be interesting to see if the new log messages allow us to diagnose why that call to `delete_volume` fails.
`Volume vol-a346f1b3 is currently attached to i-115ccfe6`
Why is that volume still attached?
Is it because we haven't yet attempted to detach it?
No, I can see a "successful" detach operation logged prior to calling delete, so perhaps we need to make the detach method more robust.

```
2015-05-27 00:24:17+0000 [-] ELIOT: {u'action_status': u'started',
	 u'action_type': u'flocker:node:agents:blockdevice:aws',
	 'operation': ['detach_volume', (u'vol-a346f1b3',), {}],
	 u'task_level': [1],
	 u'task_uuid': u'3be1d055-b5a4-48b8-ac58-4326b8e79555',
	 u'timestamp': 1432686257.168922}
2015-05-27 00:24:17+0000 [-] ELIOT: {u'action_status': u'succeeded',
	 u'action_type': u'flocker:node:agents:blockdevice:aws',
	 u'task_level': [2],
	 u'task_uuid': u'3be1d055-b5a4-48b8-ac58-4326b8e79555',
	 u'timestamp': 1432686257.328639}

```